gazebo_plugins/src/gazebo_ros_hand_of_god.cpp

```diff
@@ -0,0 +1,187 @@
+
+namespace gazebo_plugins
+{
+class GazeboRosHandOfGodPrivate
+{
+public:
+  /// Callback to be called at every simulation iteration.
+  void OnUpdate();
+
+  /// A pointer to the GazeboROS node.
+  gazebo_ros::Node::SharedPtr ros_node_;
+
+  /// Pointer to link.
+  gazebo::physics::LinkPtr link_;
+
+  /// TF buffer
+  std::shared_ptr<tf2_ros::Buffer> buffer_;
+
+  /// TF listener
+  std::shared_ptr<tf2_ros::TransformListener> transform_listener_;
+
+  /// To broadcast TFs
+  std::shared_ptr<tf2_ros::TransformBroadcaster> transform_broadcaster_;
+
+  /// frame ID
+  std::string frame_;
+
+  /// Applied force and torque gains
+  double kl_, ka_, cl_, ca_;
+
+  /// Connection to event called at every world iteration.
+  gazebo::event::ConnectionPtr update_connection_;
+};
+
+GazeboRosHandOfGod::GazeboRosHandOfGod()
+: impl_(std::make_unique<GazeboRosHandOfGodPrivate>())
+{
+}
+
+GazeboRosHandOfGod::~GazeboRosHandOfGod()
+{
+}
+
+void GazeboRosHandOfGod::Load(gazebo::physics::ModelPtr _model, sdf::ElementPtr _sdf)
+{
+  // Initialize ROS node
+  impl_->ros_node_ = gazebo_ros::Node::Get(_sdf);
+
+  impl_->frame_ = _sdf->Get<std::string>("frame_id", "world").first;
+
+  impl_->kl_ = _sdf->Get<double>("kl", 200).first;
+  impl_->ka_ = _sdf->Get<double>("ka", 200).first;
+
+  if (_sdf->HasElement("link_name")) {
+    auto link_name = _sdf->Get<std::string>("link_name");
+    impl_->link_ = _model->GetLink(link_name);
+    if (!impl_->link_) {
+      RCLCPP_ERROR(impl_->ros_node_->get_logger(),
+        "Link [%s] not found. Aborting", link_name.c_str());
+      impl_->ros_node_.reset();
+      return;
+    }
+  } else {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(), "Please specify <link_name>. Aborting");
+    impl_->ros_node_.reset();
+    return;
+  }
+
+  impl_->link_->SetGravityMode(false);
+
+  impl_->cl_ = 2.0 * sqrt(impl_->kl_ * impl_->link_->GetInertial()->Mass());
+  impl_->ca_ = 2.0 * sqrt(impl_->ka_ * impl_->link_->GetInertial()->IXX());
+
+  impl_->buffer_ = std::make_shared<tf2_ros::Buffer>(impl_->ros_node_->get_clock());
+  impl_->transform_listener_ = std::make_shared<tf2_ros::TransformListener>(*impl_->buffer_);
+  impl_->transform_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(impl_->ros_node_);
+
+  // Listen to the update event (broadcast every simulation iteration)
+  impl_->update_connection_ = gazebo::event::Events::ConnectWorldUpdateBegin(
+    std::bind(&GazeboRosHandOfGodPrivate::OnUpdate, impl_.get()));
+}
+
+void GazeboRosHandOfGodPrivate::OnUpdate()
+{
+  // Get TF transform relative to the /world link
+  geometry_msgs::msg::TransformStamped hog_desired_tform;
+  static bool errored = false;
+  try {
+    hog_desired_tform = buffer_->lookupTransform("world", frame_ + "_desired", tf2::TimePointZero);
+    errored = false;
+  } catch (const tf2::TransformException & ex) {
+    if (!errored) {
+      RCLCPP_WARN(ros_node_->get_logger(), "%s", ex.what());
+      errored = true;
+    }
+    return;
+  }
```

Comment:
I know this logic was ported from the ROS 1 version, but we could remove `errored` and use `RCLCPP_WARN_ONCE` instead with the same effect:

```.diff
diff --git a/gazebo_plugins/src/gazebo_ros_hand_of_god.cpp b/gazebo_plugins/src/gazebo_ros_hand_of_god.cpp
index ab21f87..ff50948 100644
--- a/gazebo_plugins/src/gazebo_ros_hand_of_god.cpp
+++ b/gazebo_plugins/src/gazebo_ros_hand_of_god.cpp
@@ -139,15 +139,10 @@ void GazeboRosHandOfGodPrivate::OnUpdate()
 {
   // Get TF transform relative to the /world link
   geometry_msgs::msg::TransformStamped hog_desired_tform;
-  static bool errored = false;
   try {
     hog_desired_tform = buffer_->lookupTransform("world", frame_ + "_desired", tf2::TimePointZero);
-    errored = false;
   } catch (const tf2::TransformException & ex) {
-    if (!errored) {
-      RCLCPP_WARN(ros_node_->get_logger(), "%s", ex.what());
-      errored = true;
-    }
+    RCLCPP_WARN_ONCE(ros_node_->get_logger(), "%s", ex.what());
     return;
   }
```